### DISCUSSION TITLE
Commands Order

### DISCUSSION THREAD

**Discussion by ammarkalo** [visose/Robots#262] [Q&A] [answered]
I'm programming a simple toolpath and DO commands before and after each set of targets which are generated from different curves.
The data output from the lists definition seems OK , with the commands in their correct spot. However when I plug them into the create program component, the first command shifts down the list by 1 for some reason, and I'm getting error. I realize the error is because I'm plugging is a list with commands into the Targets list, but I'm not sure how to order the data otherwise.

What's the right way to insert commands?

![file](https://github.com/visose/Robots/assets/121248906/410cf4cd-1367-404d-9a01-a951e995df8b)

**Comment by visose** [comment]
The data in the list looks incorrect. You seem to have Targets and Commands mixed. I presume the error you are getting is about not being able to convert Targets to Commands.

**Reply by ammarkalo** [reply]
Yes they are mixed and the error you mentioned is correct. However, I'm not sure how else to feed in the commands. The example file for pick and place has a similar setup for IOs with the two mixed.
Whats the proper way of doing it? 

[Basic DO.zip](https://github.com/visose/Robots/files/14000476/Basic.DO.zip)

**Reply by visose** [reply]
This is a Grasshopper question rather than a Robots plugin question.
For Grasshopper questions you'll get better help in: https://discourse.mcneel.com/
But in summary, you need to have two separate lists, one with only Targets and one with only Commands. There should be a 1 to 1 relationship (the lists should have the same number of items). If you want to add multiple commands per target, you can use the "Group command" component to combine multiple commands into a single one.

**Comment by visose** [comment, marked as answer]
Closing this as answered. Keep Targets and Commands in separate lists with a one-to-one relationship, then feed the command list into the Command input on Create Target. If multiple commands should run at the same target, combine them first with Group Command. Do not mix Target and Command objects in the same Targets list.
